bot/engine/src/main/kotlin/connector/ConnectorQueue.kt

```diff
@@ -0,0 +1,70 @@
+package ai.tock.bot.connector
+
+import ai.tock.bot.engine.action.Action
+import ai.tock.shared.Executor
+import com.google.common.cache.Cache
+import com.google.common.cache.CacheBuilder
+import java.time.Duration
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.TimeUnit
+
+class ConnectorQueue(private val executor: Executor) {
+
+    private data class ActionWithTimestamp(val action: Action, val timestamp: Long)
+
+    private val messagesByRecipientMap: Cache<String, ConcurrentLinkedQueue<ActionWithTimestamp>> =
+        CacheBuilder.newBuilder()
+            .expireAfterAccess(1, TimeUnit.MINUTES)
+            .build()
+
+    /**
+     * Add an action to send to a queue by recipient.
+     *
+     * @param action the action to send
+     * @param delayInMs the optional delay
+     * @params
+     */
+    fun add(action: Action, delayInMs: Long, send: (action: Action) -> Unit) {
+        val id = action.recipientId.id.intern()
+
+        val actionWrapper = ActionWithTimestamp(action, System.currentTimeMillis() + delayInMs)
+
+        val queue = messagesByRecipientMap
+            .get(id) { ConcurrentLinkedQueue() }
```

Comment:
Je ne sais pas si le code s'exécute ou pas s'il n'y a pas d'ID correspondant ?

Reply:
Returns the value associated with key in this cache, obtaining that value from loader if necessary. 
The method improves upon the conventional "if cached, return; otherwise create, cache and return" pattern.

in this code my loader create an empty ConcurrentLinkedQueue

Reply:
parfait